File: websocket/model/view_model/UserVM.py
from dataclasses import dataclass, field
from typing import Optional, List

@dataclass
class UserVM():
    _id: str
    displayName: str
    displayColor: str
    profilePic: Optional[str] = None
    profileStatus: Optional[str] = None
    # guest: bool ???


# UserVM does not have:
#   - email
#   - rooms
#   - friends

File: test-client/src/models/viewModels/UserVM.ts
import { IUser } from "models/interfaces/IUser"

export interface UserVM extends IUser {
    _id: string,
    displayName: string,
    profilePic: string | undefined,
    profileStatus: string | undefined,
}


/* 
    UserVM does not have attributes:
        - email
        - friends
        - rooms
*/

